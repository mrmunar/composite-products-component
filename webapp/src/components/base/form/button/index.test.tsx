import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import Button from './';

afterEach(cleanup);

test('should take a snapshot', () => {
    const { baseElement } = render(<Button />);
    expect(baseElement).toMatchSnapshot();
});

test('to have defined label', () => {
    const { getByTestId } = render(<Button label="Test Label" />);
    expect(getByTestId('button')).toHaveTextContent("Test Label");
});

test('button is clickable', done => {
    function handleClick() {
        done();
    }
    const { getByTestId } = render(
        <Button onClick={handleClick} />
    );
    const node = getByTestId('button');
    fireEvent.click(node);
});