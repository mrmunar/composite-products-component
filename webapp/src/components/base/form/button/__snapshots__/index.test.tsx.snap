// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<body>
  <div>
    <button
      data-testid="button"
    />
  </div>
</body>
`;
